src/WebformCivicrmPreProcess.php

```diff
@@ -137,17 +139,20 @@ public function alterForm() {
     }
 
     // Search for existing contacts
+    $user_input = $this->form_state->getUserInput();
     $counts_count = count($this->data['contact']);
     for ($c = 1; $c <= $counts_count; ++$c) {
       $this->ent['contact'][$c] = wf_crm_aval($this->ent, "contact:$c", []);
       $existing_component = $this->node->getElement("civicrm_{$c}_contact_1_contact_existing");
-      // Search for contact if the user hasn't already chosen one
-      if ($existing_component && empty($submitted_contacts[$c])) {
-        $this->findContact($existing_component);
+      if ($existing_component) {
+        // Ensure that a user-entered cid overrides any default created during the page load
+        $cid_user_input = $user_input["civicrm_{$c}_contact_1_contact_existing"] ?? null;
```

Comment:
`is_int` causes this to evaluate to false, because it's dealing with a string. Replacing it with `is_numeric` works for my tests, though ideally it might check if the string represents an integer.

Reply:
When the user selects "Create New", it's not enough to just not set $cid_user_input as that results in `$this->ent['contact'][$c]` (a few lines below the above code) being set to the existing_contact default value which causes other fields on the form (cases, events, etc) to be populated for the default contact. Instead, `$this->ent['contact'][$c]` must be set to 0. It's therefore important to avoid testing `$user_input["civicrm_{$c}_contact_1_contact_existing"]` with `empty()` as we must differentiate between null (not submitted on the current page) and 0 (new contact, e.g. from URL param). The value can also be a string starting with "-" indicating that "Create New" was selected on current page.

In addition, while testing this I found another bug which occurs when the user clicks Next, then Prev, then changes the existing_contact value. All fields are visually updated (via Ajax) as expected, but when the user clicks Next again, within $form_state the locked fields retain their original values. If the user clicks Prev again, then the locked fields display those original values belonging to the default contact rather than the selected contact.

I'll update this PR to correct these issues in the next few days.